src/System.Collections.Concurrent/ref/System.Collections.Concurrent.cs

```diff
@@ -113,7 +113,7 @@ public partial class ConcurrentDictionary<TKey, TValue> : System.Collections.Gen
         void System.Collections.Generic.IDictionary<TKey, TValue>.Add(TKey key, TValue value) { }
         bool System.Collections.Generic.IDictionary<TKey, TValue>.Remove(TKey key) { throw null; }
         void System.Collections.ICollection.CopyTo(System.Array array, int index) { }
-        void System.Collections.IDictionary.Add(object key, object? value) { }
+        void System.Collections.IDictionary.Add(object key, object value) { }
```

Comment:
@safern, is this expected?

It don't think it actually matters in such an explicit interface case, because no one will call via this signature, but... it's a little strange to be undoing it.

Reply:
I mentioned this in the original PR comment. I talked to @safern about this, and his change to GenAPI (to make it nullable aware, which I used here - https://github.com/dotnet/arcade/pull/2679) removes the nullablility annotations on explicit interface implementations.

Reply:
The reason why it removes it is because we’re no longer preserving nullable metadata on non-public members, since explicit interfaces are non-public the compiler drops them, so we don’t have a way of knowing it’s nullability in GenAPI. I don’t love the idea but i can’t think of a way to preserve its nullability but by directly using the information in the interface declaration itself.

Reply:
Just to highlight explicitly, this change does cause a compiler warning which we are disabling: `CS8617`

https://github.com/dotnet/roslyn/issues/30845

cc @AlekseyTs, @gafter